### PR TITLE
Popup: Use existing popups to calculate stacking

### PR DESCRIPTION
**Pull request by Attamusc**
Fixes various issues we're seen with popup's not stacking correctly when multiple versions
of BeFF exist on a page.

### CODE REVIEW

File: View/Dialog/Popup.js [hunk 1]
```diff
@@ -21,8 +21,16 @@ define([
     },
 
     render: function() {
-      constructor.Z_INDEX += 2;
-      this._zIndex = constructor.Z_INDEX;
+      var preExistingZIndex = Number($('.popup').css('zIndex'));
```
Comment (mikesherov) [inline View/Dialog/Popup.js]: what if there is more than 1 popup?

Comment (mikesherov) [inline View/Dialog/Popup.js]: can we configure the `popup` class to look for?

Comment (Attamusc) [inline View/Dialog/Popup.js]: oh duh; should be a reduce for the highest zIndex

Comment (Attamusc) [inline View/Dialog/Popup.js]: these views get their type added to them automatically, but I can make it configurable

File: View/Dialog/Popup.js [hunk 2]
```diff
@@ -21,8 +23,19 @@ define([
     },
 
     render: function() {
-      constructor.Z_INDEX += 2;
-      this._zIndex = constructor.Z_INDEX;
+      var preExistingZIndex = $(this._selector).toArray()
+      .reduce(function(zIndex, el) {
+        var elZIndex = Number($(el).css('zIndex'));
+
+        if (typeof elZIndex === 'undefined' || isNaN(elZIndex)) {
```
Comment (mikesherov) [inline View/Dialog/Popup.js]: how can this ever be undefined?

Comment (Attamusc) [inline View/Dialog/Popup.js]: good point; only needs the nan check

File: View/Dialog/Popup.js [hunk 3]
```diff
@@ -21,8 +23,19 @@ define([
     },
 
     render: function() {
-      constructor.Z_INDEX += 2;
-      this._zIndex = constructor.Z_INDEX;
+      var preExistingZIndex = $(this._selector).toArray()
+      .reduce(function(zIndex, el) {
```
Comment (mikesherov) [inline View/Dialog/Popup.js]: better as a map, and then `Math.max.apply(Math, zIndexes)`?

File: View/Dialog/Popup.js (hunk 2 above)
Comment (mikesherov) [inline View/Dialog/Popup.js]: this one is blocker. Other isn't

File: View/Dialog/Popup.js (hunk 3 above)
Comment (Attamusc) [inline View/Dialog/Popup.js]: Hmm that might be clearer. I like that.

**Comment by mikesherov** [comment]
👍

**Comment by jrefano** [comment]
LGTM

**Pull request merged**